src/firmware/main/tick.c

```diff
@@ -118,6 +120,18 @@ static void normal_task(void *UNUSED(param))
         // Submit the log record, if we filled one.
         if (record)
         {
+            int current_primitive_index =
+                app_primitive_manager_getCurrentPrimitiveIndex(primitive_manager);
```

Comment:
getCurrentPrimitiveIndex() returns an unsigned int.

Probably won't cause any issues, but maybe?

Reply:
_and then_ we cast it to a `uint8_t` below..... 

So yeah, changed to `unsigned int`.